samples/performance/image_compression_control/image_compression_control.h

```diff
@@ -0,0 +1,189 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "rendering/postprocessing_pipeline.h"
+#include "rendering/render_pipeline.h"
+#include "scene_graph/components/camera.h"
+#include "scene_graph/components/perspective_camera.h"
+#include "vulkan_sample.h"
+
+/**
+ * @brief Image Compression Control Sample
+ *
+ * This sample shows how to use the extensions VK_EXT_image_compression_control
+ * and VK_EXT_image_compression_control_swapchain to select between
+ * different levels of image compression.
+ *
+ * The UI shows the impact compression has on image size and bandwidth,
+ * illustrating the benefits of fixed-rate (visually lossless) compression.
+ */
+class ImageCompressionControlSample : public vkb::VulkanSample
+{
+  public:
+	ImageCompressionControlSample();
+
+	virtual bool prepare(const vkb::ApplicationOptions &options) override;
+
+	virtual void request_gpu_features(vkb::PhysicalDevice &gpu) override;
+
+	virtual ~ImageCompressionControlSample() = default;
+
+	virtual void update(float delta_time) override;
+
+	virtual void render(vkb::CommandBuffer &command_buffer) override;
+
+	void draw_gui() override;
+
+  private:
+	vkb::sg::PerspectiveCamera *camera{nullptr};
+
+	virtual void prepare_render_context() override;
+
+	virtual void create_render_context() override;
+
+	std::unique_ptr<vkb::RenderTarget> create_render_target(vkb::core::Image &&swapchain_image);
+
+	enum Attachments : int
+	{
+		Swapchain = 0,
+		Depth     = 1,
+		Color     = 2,
+	};
+
+	/**
+	 * @brief Postprocessing pipeline
+	 * In this sample, a postprocessing pass applies a simple chromatic aberration effect
+	 * to the output of the forward rendering pass. Since this color output needs to
+	 * be saved to main memory, it is a simple use case for compressing the image and
+	 * saving bandwidth and memory footprint, as illustrated in the sample.
+	 */
+	std::unique_ptr<vkb::PostProcessingPipeline> postprocessing_pipeline{};
+
+	/**
+	 * @brief Load/store operations of the forward rendering pass attachments
+	 * Used to specify that the color output must be stored to main memory.
+	 */
+	std::vector<vkb::LoadStoreInfo> scene_load_store{};
+
+	/**
+	 * @brief Color image parameters
+	 * These are the properties of the color output image.
+	 */
+	VkImageCreateInfo color_image_info{VK_STRUCTURE_TYPE_IMAGE_CREATE_INFO};
+
+	/**
+	 * @brief Compression scheme
+	 * By default, many implementations support lossless compression.
+	 * Some also support fixed-rate compression, which is generally visually lossless.
+	 * This flag can also be used to disable all compression, which is not recommended.
+	 */
+	VkImageCompressionFlagsEXT compression_flag{VK_IMAGE_COMPRESSION_DEFAULT_EXT};
+
+	/**
+	 * @brief Possible compression schemes
+	 */
+	enum TargetCompression : int
```

Comment:
Maybe use an `enum class` here, to prevent polluting the namespace with the enum values.

Reply:
Done, although it meant adding many casts, wouldn't this have been cleaner?